File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/spotcleaning/SpotCleaningBasic1Service.java
package com.neatorobotics.sdk.android.robotservices.spotcleaning;

/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public class SpotCleaningBasic1Service extends SpotCleaningService {

    private static final String TAG = "SpotCleaningBasic1";

    @Override
    public boolean isEcoModeSupported() {
        return true;
    }

    @Override
    public boolean isTurboModeSupported() {
        return true;
    }

    @Override
    public boolean isExtraCareModeSupported() {
        return false;
    }

    @Override
    public boolean isCleaningAreaSupported() {
        return true;
    }

    @Override
    public boolean isCleaningFrequencySupported() {
        return true;
    }

    @Override
    public boolean isFloorPlanSupported() {
        return false;
    }

    @Override
    public boolean areZonesSupported() {
        return false;
    }

}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/spotcleaning/SpotCleaningBasic1Service.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.spotcleaning

import com.neatorobotics.sdk.android.models.CleaningMode
import com.neatorobotics.sdk.android.models.CleaningModifier
import com.neatorobotics.sdk.android.models.NavigationMode

class SpotCleaningBasic1Service : SpotCleaningService() {

    override val isMultipleZonesCleaningSupported: Boolean
        get() = false

    override val isEcoModeSupported: Boolean
        get() = true

    override val isTurboModeSupported: Boolean
        get() = true

    override val isExtraCareModeSupported: Boolean
        get() = false

    override val isCleaningAreaSupported: Boolean
        get() = true

    override val isCleaningFrequencySupported: Boolean
        get() = true

    init {
        this.cleaningMode = CleaningMode.ECO
        this.cleaningModifier = CleaningModifier.DOUBLE
        this.cleaningNavigationMode = NavigationMode.NORMAL
    }

    companion object {

        private const val TAG = "SpotCleaningBasic1"
    }

}
